vertical-pod-autoscaler/pkg/admission-controller/logic/resource_handler.go

```diff
@@ -0,0 +1,35 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package logic
+
+import (
+	"k8s.io/api/admission/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/autoscaler/vertical-pod-autoscaler/pkg/utils/metrics/admission"
+)
+
+// ResourceHandler represents a handler for a resource in Admission Server
+type ResourceHandler interface {
```

Comment:
I think we don't need to make this interface visible outside vertical-pod-autoscaler/pkg/admission-controller/logic/server.go

Reply:
We need. The goal of this change is to be able to inject custom controllers into this Admission Server, which I will do in a next step.

Reply:
I think we don't need it. Even if an interface is private part of a module you can still implement the niterface outside the module